examples/simple_example.rs

```diff
@@ -65,36 +64,53 @@ fn tracer() -> Box<Tracer> {
     }
 }
 
+/// Prints the addresses of the first `qty` blocks in a trace along with it's name and
+/// computation result.
+fn print_trace(trace: &Box<Trace>, name: &str, result: u32, qty: usize) {
+    let count = trace.iter_blocks().count();
+    println!("{}: num_blocks={}, result={}", name, count, result);
+    let mut itr = trace.iter_blocks();
+    let mut end = false;
+    for i in 0..qty {
```

Comment:
I think you can simplify most of the code in this function down to something like:

```rust
trace.iter_blocks().take(qty).map(|x| println!("  block ...", ...));
if qty > count {
  println!("  ... {} more", ...);
}
```

Reply:
Sounds good.

Reply:
Is there a better way than using `last()` to consume the iterator?

Reply:
For context:
```
warning: unused `std::iter::Map` which must be used: iterator adaptors are lazy and do nothing unless consumed
  --> examples/simple_example.rs:73:5
   |
73 | /     trace.iter_blocks().take(qty)
74 | |                        .enumerate()
75 | |                        .map(|(i, blk)| println!("  block {}: 0x{:x}", i, blk.unwrap().start_vaddr()));
   | |______________________________________________________________________________________________________^
   |
```

Reply:
Good point. This should probably be a `for` loops in retrospect:

```rust
for x in trace.iter_blocks().take(qty) {
  // do something with x
}
```

Reply:
Using `.last()` works fine. I think with a comment I actually prefer what you were proposing.

```diff
diff --git a/examples/simple_example.rs b/examples/simple_example.rs
index 2260d51..6b083a9 100644
--- a/examples/simple_example.rs
+++ b/examples/simple_example.rs
@@ -69,28 +69,24 @@ fn tracer() -> Box<Tracer> {
 fn print_trace(trace: &Box<Trace>, name: &str, result: u32, qty: usize) {
     let count = trace.iter_blocks().count();
     println!("{}: num_blocks={}, result={}", name, count, result);
-    let mut itr = trace.iter_blocks();
-    let mut end = false;
-    for i in 0..qty {
-        if let Some(blk) = itr.next() {
-            println!("  block {}:\t0x{:x}", i, blk.unwrap().start_vaddr());
-        } else {
-            end = true;
-        }
-    }
-    if !end {
+
+    trace.iter_blocks().take(qty)
+                       .enumerate()
+                       .map(|(i, blk)| println!("  block {}: 0x{:x}", i, blk.unwrap().start_vaddr()))
+                       .last();
+    if count > qty {
         println!("  ... {} more", count - qty);
     }
     println!("");
 }
```

Reply:
The Rust docs say (and I was being an idiot for not thinking of this) "The idiomatic way to write a map for its side effects is to use a for loop instead".

Reply:
Fair enough. I'll see what I can come up with.